src/sentry/api/serializers/models/organization.py

```diff
@@ -116,24 +117,38 @@ def serialize(self, obj, attrs, user):
 
 
 class DetailedOrganizationSerializer(OrganizationSerializer):
-    def serialize(self, obj, attrs, user):
+    def get_attrs(self, item_list, user, **kwargs):
+        return super(DetailedOrganizationSerializer, self).get_attrs(item_list, user)
+
+    def serialize(self, obj, attrs, user, access):
         from sentry import experiments
-        from sentry.app import env
         from sentry.api.serializers.models.project import ProjectSummarySerializer
         from sentry.api.serializers.models.team import TeamSerializer
 
-        team_list = sorted(Team.objects.filter(
-            organization=obj,
-            status=TeamStatus.VISIBLE,
-        ), key=lambda x: x.slug)
+        # If the current user has global access but is not on any
+        # teams show them all teams. This covers
+        # the superuser scenario.
+        if access.has_global_access and len(access.teams) == 0:
+            team_list = Team.objects.filter(
```

Comment:
Will we actually be doing this for multiple organizations? I imagine mostly it's just single orgs, but if it is many then we're adding extra queries by doing these checks here vs `get_attrs`

Reply:
I've only seen this serializer used in the context of multiple organizations. It only appears to be used for single orgs.